src/pages/Search/components/SearchNavigationBar.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import styles from '../Search.module.scss';
+
+// eslint-disable-next-line max-len
+export default function SearchNavigationBar({ children }: React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>) {
+  return (
+    <nav className={styles['search-nav']}>
+      {children}
+    </nav>
+  );
+}
```

Comment:
1. DetailedHTMLProps 가 뭔가 하고 봤는데 class component 를 지원하기 위한 타입같아요. 굳이 안써줘도 될것 같아요.
```tsx
type DetailedHTMLProps<E extends HTMLAttributes<T>, T> = ClassAttributes<T> & E;
```
2. nav tag 가 컴포넌트의 root tag 인데 HTMLDivElement 를 generic 으로 받는게 어색해요. 
3. props 로 HTMLAttributes 를 다 받을거면 그냥 spread 해줘도 될것 같아요. 
아니면 props 로 children 만 받도록  `{ children?: React.ReactNode }` 를 받는게 좋을것 같아요.

```suggestion
export default function SearchNavigationBar(props: React.HTMLAttributes<HTMLElement>) {
  return (
    <nav className={styles['search-nav']} {...props} />
  );
}
```

Reply:
> props 로 HTMLAttributes 를 다 받을거면 그냥 spread 해줘도 될것 같아요.
> 아니면 props 로 children 만 받도록 { children?: React.ReactNode } 를 받는게 좋을것 같아요.

전자의 경우는 `<SearchNavigationBar></SearchNavigationBar>` 에 자식 요소로 넣을 때 HTML 속성값들도 같이 받을 수 있다는건가요?!?
후자의 경우는 그냥 요소만 받도록 하는거구요? 🤔

Reply:
`    <nav className={styles['search-nav']} {...props} />
`

이렇게 요소의 props로 넘겨주는 방법은 몰랐네요.. 이런 방법도 있구나 싶었습니다

Reply:
> > props 로 HTMLAttributes 를 다 받을거면 그냥 spread 해줘도 될것 같아요.
> > 아니면 props 로 children 만 받도록 { children?: React.ReactNode } 를 받는게 좋을것 같아요.
> 
> 전자의 경우는 `<SearchNavigationBar></SearchNavigationBar>` 에 자식 요소로 넣을 때 HTML 속성값들도 같이 받을 수 있다는건가요?!? 후자의 경우는 그냥 요소만 받도록 하는거구요? 🤔

넵 좀 더 정확히는... 전자의 경우는 React 에서 nav tag 를 사용했을 때 지원하는 property (HTMLElement) 를 모두 사용할 수 있는거고요 후자는 `childeren` prop 만 받게 하는거예요

Reply:
그런거였군요..! 후자만 사용하다보니 뭔가 전자의 경우는 아직까진 확 와 닿지 않네요... ㅜㅜ
 그러면 props를 spread 해줬을 때 HTML 속성값도 사용할 수 있는건데 이 경우 보편적인 예시가 있을까요?

Reply:
가령 이벤트 핸들러를 바로 붙일 수 있죠 :)